tracks/swift/exercises/difference-of-squares/mentoring.md

```diff
@@ -0,0 +1,54 @@
+# Difference of Squares
+
+_Differences of Squares_ is a core exercise, unlocked after _Gigasecond_.
+
+## Reasonable Solutions
+
+```swift
+class Squares {
+    let maxNum: Int
+    
+    lazy var squareOfSum: Int = {
+        let sum = (0...self.maxNum).reduce(0, { $0 + $1 })
+        return sum.squared
+    }()
+    
+    lazy var sumOfSquares: Int = {
+        return (0...self.maxNum).reduce(0, { $0 + $1.squared })
+    }()
+    
+    lazy var differenceOfSquares: Int = {
+        return self.squareOfSum - self.sumOfSquares
+    }()
+    
+    init(_ maxNum: Int) {
```

Comment:
It looks like the exercise requires the creation of an object that stores `maxNum`; I've seen similar use of OO in the Perl track.

So without modifying the exercise, this appears to be the right thing to do. But I don't think that makes it "reasonable".

Have you considered changing the exercise so it doesn't wrap integers in classes just to perform method calls on them?

(This comment is not a request to change anything in this PR.)

Reply:
Yes, but in https://github.com/exercism/swift

The reviewer there will be a track maintainer.

Tracks can have conventions of their own.